src/representations/calculator_base.hh

```diff
@@ -50,51 +50,70 @@ namespace rascal {
 
     //! Move constructor
-    RepresentationManagerBase(RepresentationManagerBase && other) = default;
+    CalculatorBase(CalculatorBase && other) = default;
 
     //! Destructor
-    virtual ~RepresentationManagerBase() = default;
+    virtual ~CalculatorBase() = default;
 
     //! Copy assignment operator
-    RepresentationManagerBase &
-    operator=(const RepresentationManagerBase & other) = delete;
+    CalculatorBase & operator=(const CalculatorBase & other) = delete;
 
     //! Move assignment operator
-    RepresentationManagerBase &
-    operator=(RepresentationManagerBase && other) = default;
+    CalculatorBase & operator=(CalculatorBase && other) = default;
 
     //! Pure Virtual Function to set hyperparameters of the representation
     virtual void set_hyperparameters(const Hypers_t &) = 0;
 
-    //! Pure Virtual Function to set hyperparameters of the representation
+    //!
     void check_hyperparameters(const ReferenceHypers_t &, const Hypers_t &);
 
-    //! Compute the representation using a StructureManager
-    virtual void compute() = 0;
-
-    //! get the raw data of the representation
-    virtual std::vector<Precision_t> & get_representation_raw_data() = 0;
-
-    virtual Data_t & get_representation_sparse_raw_data() = 0;
-
-    //! get the size of a feature vector
-    virtual size_t get_feature_size() = 0;
-
-    //! get the number of centers for the representation
-    virtual size_t get_center_size() = 0;
+    //! return the name of the calculator
+    inline const std::string & get_name() const { return this->name; }
+
+    //! return the name of the calculator's gradients
+    inline std::string get_gradient_name() const {
+      return this->name + std::string("_gradients");
+    }
+
+    inline const std::string & get_prefix() const {
+      return this->default_prefix;
+    }
+
+    //! set the name of the calculator
+    inline void set_name(const std::string & name) { this->name = name; }
+    //! set the prefix for the default naming of the representation
+    inline void set_default_prefix(const std::string & default_prefix) {
+      this->default_prefix = default_prefix;
+    }
+    /**
+     * identifier is a user defined name for the representation used to
+     * register the computed representation
+     * if not provided use the hypers to generate a unique identifier
+     */
+    inline void set_name(const Hypers_t & hyper) {
+      if (hyper.count("identifier") == 1) {
+        this->set_name(this->default_prefix +
+                       hyper["identifier"].get<std::string>());
+      } else {
+        this->set_name(this->default_prefix + hyper.dump());
+      }
+    }
+
+    /**
+     * Computes the representation associated to the input structure
+     * manager. It is templated so it can't be virtual but it is still
+     * expected.
+     */
+    // template<class StructureManager>
+    // virtual void compute(StructureManager& ) = 0;
```

Comment:
We should maybe consider to use at some point CRTP for the calculator to make templatable virtual functions for `compute` and `compute_impl`
1. to avoid the repetition of the `compute_loop` function in each representation
2. It is easier for non rascal developer to write a representation
3. to give immediately compiler errors when not implemented (however the later error at the binding level is also informative enough)
However this makes the code harder to read at other points, especially to make this work for variadic enum classes takes some extra template structures as in the structure manager to build the cluster_indices_container type. Nothing for this pull request, but should I do a issue for improvement for this?

Reply:
added as issue for future, not important now